addons/mail/models/discuss/mail_guest.py

```diff
@@ -2,12 +2,60 @@
 
 import pytz
 import uuid
+from functools import wraps
+from inspect import Parameter, signature
 
 from odoo.tools import consteq
 from odoo import _, api, fields, models
+from odoo.http import request
 from odoo.addons.base.models.res_partner import _tz_get
 from odoo.exceptions import UserError
 from odoo.addons.bus.models.bus_presence import AWAY_TIMER, DISCONNECTION_TIMER
+from odoo.addons.bus.websocket import wsrequest
+
+
+def add_guest_to_context(func):
+    """ Decorate a function to extract the guest from the request.
+    The guest is then available on the context of the current
+    request.
+    """
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        req = request or wsrequest
+        token = (
+            kwargs.pop("guest_token", None)
+            or req.httprequest.cookies.get(req.env["mail.guest"]._cookie_name)
+            or req.env.context.get("guest_token", "")
+        )
+        parts = token.split(req.env["mail.guest"]._cookie_separator)
+        guest = req.env.context.get("guest", req.env["mail.guest"])
+        if len(parts) == 2:
+            guest_id, guest_access_token = parts
+            guest = req.env["mail.guest"].browse(int(guest_id)).sudo().exists()
```

Comment:
The retrieved guest will be in sudo mode, right? Shouldn't we `sudo(False)` it before putting it in the context?

Reply:
Indeed, sudo mode was disabled when calling `_get_guest_from_request`. I'll restore this behavior here as wel.